### ISSUE TITLE
Automatically enable GPU support on Windows

### ISSUE THREAD

**Issue by gchindemi** [BelloneLab/lisbet#2]
Installing pytorch on Windows using pip will get the CPU version by default.

For now, users can get the GPU version by running
`pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124`
after installing LISBET, but this should be the default behavior.

**Comment by kipkeller** [comment]
Sorry - Windows 10 PC with NVIDIA GetForce RTX 4070 Ti

**Comment by kipkeller** [comment]
Tried python=3.12 without any success.
Here's my list (and the code I am trying to run):
aiohappyeyeballs          2.6.1                    pypi_0    pypi
aiohttp                   3.12.13                  pypi_0    pypi
aiosignal                 1.3.2                    pypi_0    pypi
attrs                     25.3.0                   pypi_0    pypi
bottleneck                1.5.0                    pypi_0    pypi
bzip2                     1.0.8                h2bbff1b_6
ca-certificates           2025.2.25            haa95532_0
cartopy                   0.24.1                   pypi_0    pypi
certifi                   2025.6.15                pypi_0    pypi
cftime                    1.6.4.post1              pypi_0    pypi
charset-normalizer        3.4.2                    pypi_0    pypi
colorama                  0.4.6                    pypi_0    pypi
contourpy                 1.3.2                    pypi_0    pypi
crc32c                    2.7.1                    pypi_0    pypi
cycler                    0.12.1                   pypi_0    pypi
donfig                    0.8.1.post1              pypi_0    pypi
expat                     2.7.1                h8ddb27b_0
filelock                  3.18.0                   pypi_0    pypi
flox                      0.10.4                   pypi_0    pypi
fonttools                 4.58.4                   pypi_0    pypi
frozenlist                1.7.0                    pypi_0    pypi
fsspec                    2025.5.1                 pypi_0    pypi
h5netcdf                  1.6.1                    pypi_0    pypi
h5py                      3.14.0                   pypi_0    pypi
hdmf                      4.1.0                    pypi_0    pypi
hmmlearn                  0.3.3                    pypi_0    pypi
huggingface-hub           0.33.0                   pypi_0    pypi
idna                      3.10                     pypi_0    pypi
imageio                   2.37.0                   pypi_0    pypi
imageio-ffmpeg            0.6.0                    pypi_0    pypi
jinja2                    3.1.6                    pypi_0    pypi
joblib                    1.5.1                    pypi_0    pypi
jsonschema                4.24.0                   pypi_0    pypi
jsonschema-specifications 2025.4.1                 pypi_0    pypi
kiwisolver                1.4.8                    pypi_0    pypi
libffi                    3.4.4                hd77b12b_1
lightning                 2.5.1.post0              pypi_0    pypi
lightning-utilities       0.14.3                   pypi_0    pypi
lisbet                    0.3.1                    pypi_0    pypi
llvmlite                  0.44.0                   pypi_0    pypi
loguru                    0.7.3                    pypi_0    pypi
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                3.0.2                    pypi_0    pypi
matplotlib                3.10.3                   pypi_0    pypi
mdurl                     0.1.2                    pypi_0    pypi
movement                  0.7.1                    pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
multidict                 6.5.0                    pypi_0    pypi
napari-video              0.2.10                   pypi_0    pypi
nc-time-axis              1.4.1                    pypi_0    pypi
ndx-pose                  0.2.2                    pypi_0    pypi
netcdf4                   1.7.2                    pypi_0    pypi
networkx                  3.5                      pypi_0    pypi
numba                     0.61.2                   pypi_0    pypi
numbagg                   0.9.0                    pypi_0    pypi
numcodecs                 0.16.1                   pypi_0    pypi
numpy                     2.2.6                    pypi_0    pypi
numpy-groupies            0.11.3                   pypi_0    pypi
opencv-python-headless    4.11.0.86                pypi_0    pypi
openssl                   3.0.16               h3f729d1_0
opt-einsum                3.4.0                    pypi_0    pypi
packaging                 24.2                     pypi_0    pypi
pandas                    2.3.0                    pypi_0    pypi
pillow                    11.2.1                   pypi_0    pypi
pip                       25.1               pyhc872135_2
platformdirs              4.3.8                    pypi_0    pypi
pooch                     1.8.2                    pypi_0    pypi
propcache                 0.3.2                    pypi_0    pypi
pygments                  2.19.1                   pypi_0    pypi
pynndescent               0.5.13                   pypi_0    pypi
pynwb                     3.0.0                    pypi_0    pypi
pyparsing                 3.2.3                    pypi_0    pypi
pyproj                    3.7.1                    pypi_0    pypi
pyshp                     2.3.1                    pypi_0    pypi
python                    3.12.11              h716150d_0
python-dateutil           2.9.0.post0              pypi_0    pypi
pytorch-lightning         2.5.1.post0              pypi_0    pypi
pytz                      2025.2                   pypi_0    pypi
pyvideoreader             0.5.7                    pypi_0    pypi
pyyaml                    6.0.2                    pypi_0    pypi
qt-niu                    0.1.0                    pypi_0    pypi
qtpy                      2.4.3                    pypi_0    pypi
referencing               0.36.2                   pypi_0    pypi
requests                  2.32.4                   pypi_0    pypi
rich                      14.0.0                   pypi_0    pypi
rpds-py                   0.25.1                   pypi_0    pypi
ruamel-yaml               0.18.14                  pypi_0    pypi
ruamel-yaml-clib          0.2.12                   pypi_0    pypi
scikit-learn              1.7.0                    pypi_0    pypi
scipy                     1.15.3                   pypi_0    pypi
seaborn                   0.13.2                   pypi_0    pypi
setuptools                78.1.1          py312haa95532_0
shapely                   2.1.1                    pypi_0    pypi
simplejson                3.20.1                   pypi_0    pypi
six                       1.17.0                   pypi_0    pypi
sleap-io                  0.3.0                    pypi_0    pypi
sqlite                    3.45.3               h2bbff1b_0
superqt                   0.7.4                    pypi_0    pypi
sympy                     1.14.0                   pypi_0    pypi
threadpoolctl             3.6.0                    pypi_0    pypi
tk                        8.6.14               h5e9d12e_1
toolz                     1.0.0                    pypi_0    pypi
torch                     2.7.1+cu118              pypi_0    pypi
torchaudio                2.7.1+cu118              pypi_0    pypi
torchinfo                 1.8.0                    pypi_0    pypi
torchmetrics              1.7.3                    pypi_0    pypi
torchvision               0.22.1                   pypi_0    pypi
tqdm                      4.67.1                   pypi_0    pypi
typing-extensions         4.14.0                   pypi_0    pypi
tzdata                    2025.2                   pypi_0    pypi
umap-learn                0.5.7                    pypi_0    pypi
urllib3                   2.4.0                    pypi_0    pypi
vc                        14.42                haa95532_5
vs2015_runtime            14.42.34433          hbfb602d_5
wheel                     0.45.1          py312haa95532_0
win32-setctime            1.2.0                    pypi_0    pypi
xarray                    2025.6.1                 pypi_0    pypi
xz                        5.6.4                h4754444_1
yarl                      1.20.1                   pypi_0    pypi
zarr                      3.0.8                    pypi_0    pypi
zlib                      1.2.13               h8cc25b3_1

(Lisbet02) D:\Lisbet>betman segment_motifs ^
More?    --output_path=hmm_predictions ^
More?    --min_n_components=2 ^
More?    --max_n_components=32 ^
More?    --num_iter=1000 ^
More?    -v ^
More?    embeddings

**Comment by gchindemi** [comment]
Thank you for the detailed feedback ! I'll get back to you ASAP

**Comment by gchindemi** [comment]
Ah! I just noticed that you were running `segment_motifs`.

The current version of this algorithm is not GPU-accelerated, although the different models are trained in parallel using multiprocessing by default.

In our internal testing, we found that using multiprocessing and setting `—fit_frac` to reduce the number of training data points was sufficient to achieve a good balance between quality and performance. Therefore, we haven’t explored other options for segmentation yet...

**Comment by gchindemi** [comment]
It turns out that [PyTorch has deprecated their Anaconda channel](https://github.com/pytorch/pytorch/issues/138506). Considering this, I believe the most suitable approach to support MS Windows installations using `conda` is to create the environment and then use pip to install PyTorch and LISBET.

We have updated the documentation with instructions on how to install LISBET using `conda` on MS Windows. Additionally, we have added a note in the documentation to clarify that `segment_motifs` is not GPU-accelerated.

Given these developments, I propose closing this ticket. Hopefully, future releases of PyTorch will simplify MS Windows support.
